src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -444,33 +458,195 @@ static void _repeat(LottieGroup* parent, Shape* path, RenderContext* ctx)
 }
 
 
-static void _appendRect(Shape* shape, float x, float y, float w, float h, float r, Matrix* transform, bool clockwise)
+static bool _intersect(const Line& line1, const Line& line2, Point& intersection, bool& inside)
+{
+    auto a1 = line1.pt2.y - line1.pt1.y;
+    auto b1 = line1.pt1.x - line1.pt2.x;
+    auto a2 = line2.pt2.y - line2.pt1.y;
+    auto b2 = line2.pt1.x - line2.pt2.x;
+
+    auto det = a1 * b2 - a2 * b1;
+    if (fabsf(det) < 1e-3f) return false;
+
+    auto c1 = a1 * line1.pt1.x + b1 * line1.pt1.y;
+    auto c2 = a2 * line2.pt1.x + b2 * line2.pt1.y;
+
+    intersection.x = (b2 * c1 - b1 * c2) / det;
+    intersection.y = (a1 * c2 - a2 * c1) / det;
+
+    constexpr float epsilon = 1e-3f;
+    inside = std::min(line1.pt1.x, line1.pt2.x) - epsilon <= intersection.x && intersection.x <= std::max(line1.pt1.x, line1.pt2.x) + epsilon &&
+             std::min(line1.pt1.y, line1.pt2.y) - epsilon <= intersection.y && intersection.y <= std::max(line1.pt1.y, line1.pt2.y) + epsilon &&
+             std::min(line2.pt1.x, line2.pt2.x) - epsilon <= intersection.x && intersection.x <= std::max(line2.pt1.x, line2.pt2.x) + epsilon &&
+             std::min(line2.pt1.y, line2.pt2.y) - epsilon <= intersection.y && intersection.y <= std::max(line2.pt1.y, line2.pt2.y) + epsilon;
+
+    return true;
+}
+
+
+static Line _offset(const Point& p1, const Point& p2, float offset)
+{
+    auto direction = p2 - p1;
+    auto directionUnit = direction / length(direction);
+    Point scaledNormal = {-directionUnit.y * offset, directionUnit.x * offset};
+    return {p1 - scaledNormal, p2 - scaledNormal};
+}
+
+
+static void _appendOffsetPath(Shape* shape, const PathCommand* cmds, uint32_t cmdsCnt, const Point* pts, TVG_UNUSED uint32_t ptsCnt, const RenderOffsetPath* offsetPath)
```

Comment:
this is a general solution. for some basic shapes like rectangles/circles, the size of the path after applying the offset is know, so I can still use static arrays instead of Array - need to inwestigate